contribs/gmf/src/services/permalink.js

```diff
@@ -921,16 +1025,32 @@ gmf.Permalink.prototype.getLayerStateParamFromNode_ = function(layerNode) {
 /**
  * @param {string} layerName The name of the layer.
  * @param {boolean} isMerged Whether the layer is merged or not.
+ * @param {string=} opt_propertyName Whether we are looking for a layer property value
+ * (e.g opacity)
  * @return {string} The state param for the layer
  * @private
  */
 gmf.Permalink.prototype.getLayerStateParam_ = function(layerName,
-    isMerged) {
+    isMerged, opt_propertyName) {
   var param;
   if (isMerged) {
-    param = gmf.PermalinkParamPrefix.TREE_GROUP_LAYERS + layerName;
+    switch (opt_propertyName) {
+      case gmf.PermalinkOpenLayersLayerProperties.OPACITY:
+        param = gmf.PermalinkParamPrefix.TREE_GROUP_OPACITY + layerName;
+        break;
+      default:
+        param = gmf.PermalinkParamPrefix.TREE_GROUP_LAYERS + layerName;
+        break;
+    }
   } else {
-    param = gmf.PermalinkParamPrefix.TREE_ENABLE + layerName;
+    switch (opt_propertyName) {
+      case gmf.PermalinkOpenLayersLayerProperties.OPACITY:
+        param = gmf.PermalinkParamPrefix.TREE_OPACITY + layerName;
+        break;
+      default:
+        param = gmf.PermalinkParamPrefix.TREE_ENABLE + layerName;
+        break;
+    }
   }
   return param;
```

Comment:
Since you use "+ layerName" for each case, you could do it only once in the return:

return param + layerName;